lake/index/writer.go

```diff
@@ -35,8 +36,8 @@ func NewWriter(ctx context.Context, u iosrc.URI, def *Definition) (*Writer, erro
 }
 
 type Writer struct {
-	Definition *Definition
-	URI        iosrc.URI
+	Reference Reference
```

Comment:
`Reference` is big. Use a pointer.
```suggestion
	Reference *Reference
```